torch/csrc/api/include/torch/optim/serialize.h

```diff
@@ -40,6 +127,52 @@ void serialize(
     const std::string& key,
     std::vector<int64_t>& steps);
 
+// Utility function to save state and param_groups
+template <typename DerivedOptimizerParamState, typename DerivedOptimizerParamOptions>
+void serialize(
+    serialize::OutputArchive& archive,
+    const detail::OptimizerBase& optimizer) {
+  archive.write("pytorch_version", IValue("1.5.0"));
+  serialize::OutputArchive state_archive(archive.compilation_unit());
+  detail::serialize<DerivedOptimizerParamState>(state_archive, optimizer.state());
+  archive.write("state", state_archive);
+
+  serialize::OutputArchive param_groups_archive(archive.compilation_unit());
+  detail::serialize<DerivedOptimizerParamOptions>(param_groups_archive, optimizer.param_groups());
+  archive.write("param_groups", param_groups_archive);
+}
+
+// Utility function to load state and param_groups and update state
+template <typename DerivedOptimizerParamState, typename DerivedOptimizerParamOptions>
+void serialize(
+    serialize::InputArchive& archive,
+    detail::OptimizerBase& optimizer) {
+
+    IValue pytorch_version;
+    archive.try_read("pytorch_version", pytorch_version);
```

Comment:
nit: I think we should use `archive.read` instead of `archive.try_read` here, because the former throws an error if the key is not found.